tools/setup_sbuild.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/bash
+
+USAGE="
+BE VERY CAREFUL!
+When using the \"destroy\" operation, this will run:
+
+    sudo rm /etc/schroot/chroot.d/\$release-\$arch
+    sudo rm -rf /var/lib/schroot/chroots/\$release-\$arch
+
+Based on the RELEASES and ARCHS environment vars.
+So bad environment variables could corrupt your system.
+
+When using the \"create\" operation, this will create or modify chroots on your
+system with names matching \"\$release-\$arch\" based on the RELEASES and ARCHS
+environment vars.
+
+Example Usage:
+env RELEASES=\"xenial bionic\" ARCHS=\"amd64\" OPERATION=\"create\" bash tools/setup_sbuild.sh
+"
+
+if [ -z "$RELEASES" ]; then
+  echo "please set RELEASES"
+  echo "$USAGE"
+  exit 1
+fi
+if [ -z "$ARCHS" ]; then
+  echo "please set ARCHS"
+  echo "$USAGE"
+  exit 1
+fi
+if [ -z "$OPERATION" ]; then
+  echo "please set OPERATION to \"create\" or \"destroy\""
+  echo "$USAGE"
+  exit 1
+fi
+
+set -x
+
+create_sbuild_chroot() {
+    release=$1
+    arch=$2
+    echo "creating $release $arch"
+    name="$release-$arch"
+    sudo sbuild-launchpad-chroot create --architecture="$arch" "--name=$name" "--series=$release"
+    sudo schroot -c source:$name -u root -d / -- sh -c "grep -q updates /etc/apt/sources.list || echo \"deb http://archive.ubuntu.com/ubuntu/ $release-updates main restricted universe multiverse\" >> /etc/apt/sources.list"
```

Comment:
Those commands will fail if for some reason we have another chroot with an alias of `$series-$arch`.
However, I don't know if we can actually address this here

Reply:
Yeah these are great points. I think you're right that the correct way to do this would be to expose a `chroot` argument to the build.py script, as well as the integration tests.

Something about the current setup that we should try to keep, though, is that ~97% of sbuilds on our dev machines are test builds and should use the `ua-$release-$arch` chroot. So I think the default sbuild for both `build.sh` and integration tests should be to use `ua-$release-$arch`. Then a `chroot` argument can be used to select the "standard" chroot when needed (i.e. probably for jenkins runs).

So in example form, my suggestion is:
```
./tools/setup_sbuild.sh
```
will create  `ua-$release-$arch`

And
```
./tools/build.sh xenial
tox -e behave-lxd-16.04
```
will both use `ua-$release-$arch`

Then a new argument will be added so that something like
```
./tools/build.sh xenial --chroot=xenial-amd64
env UACLIENT_BEHAVE_SBUILD_CHROOT=xenial-amd64 tox -e behave-lxd-16.04
```
can be used to use the standard chroot

What do you think @paride @lucasmoura ?

Reply:
I think that makes total sense. +1

Reply:
Makes sense. The other possible approach I see is making `build.py` check if `ua-$release-$arch` exists, in this case pass it to `sbuild` via `--chroot`. If it doesn't exist, no `--chroot` is specified and sbuild will go with its defaults.

This has the advantage of making `build.py` work out of the box without the specia `ua-` chroot.

To check if the chroot exists we can grep the output of `schroot --list` for the right `source:` chroot. I know no better way.